presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -269,6 +272,9 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                         ImmutableList.of(new com.facebook.presto.sql.planner.optimizations.EliminateCrossJoins()), // This can pull up Filter and Project nodes from between Joins, so we need to push them down again
                         ImmutableSet.of(new EliminateCrossJoins())),
                 new PredicatePushDown(metadata, sqlParser),
+                new IterativeOptimizer(
+                        stats,
+                        new PickTableLayout(metadata, sqlParser, true).rules()),
```

Comment:
Is it needed here? Shortly `AddExchanges` will set a proper table layout, in single node mode you do not overwrite it anyway.

Reply:
join enumeration would go after this.  We need the right table layout.